### PR TITLE
Convert to typescript

### PR DESCRIPTION
**Pull request by rictic**
Just realized I never sent this out.

Mostly just does the obvious thing. Most interesting changes:

- all callbacks converted to promises
- lazily copied generated type definitions for wct into custom_typings. What's the better way of doing this?
- converted gulpfile and friends to use tools-common

### CODE REVIEW

**Comment by justinfagnani** [comment]
Instead of copying typings into custom_typings, have WCT put them into a declaration file that's referenced from package.json's typings field. To do this we'd need to release WCT.

**Comment by rictic** [comment]
Mind if I punt on clean integration with wct's declarations for this PR? It should be possible with some gulp hacking, but I'd like to land this change so that I can land the change that gets wct's integration tests working again first.

File: custom_typings/cleankill.d.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+declare module 'cleankill' {
+  interface InterruptHandler {
+    (done: (message?: string) => void): void;
```
Comment (justinfagnani) [inline custom_typings/cleankill.d.ts]: Didn't this change to Promises?

File: src/browsers.ts [hunk 2]
```diff
@@ -39,67 +44,65 @@ function normalize(browsers) {
  * @param {!Array<string>} names
  * @param {function(*, Array<!Object>)} done
  */
-function expand(names, done) {
+export async function expand(names: string[]) {
```
Comment (justinfagnani) [inline src/browsers.ts]: return type?

File: src/browsers.ts [hunk 3]
```diff
@@ -24,9 +26,12 @@ var LAUNCHPAD_TO_SELENIUM = {
  * @param {Array<string|!Object>} browsers
  * @return {!Array<string>}
  */
-function normalize(browsers) {
+export function normalize(browsers: (string | {browserName: string})[]) {
```
Comment (justinfagnani) [inline src/browsers.ts]: Return type?

File: src/browsers.ts [hunk 4]
```diff
@@ -24,9 +26,12 @@ var LAUNCHPAD_TO_SELENIUM = {
  * @param {Array<string|!Object>} browsers
```
Comment (justinfagnani) [inline src/browsers.ts]: Do you need these?

File: custom_typings/cleankill.d.ts (hunk 1 above)
Comment (rictic) [inline custom_typings/cleankill.d.ts]: I haven't even gotten the typescriptified release of cleankill up on npm yet (blocked on https://github.com/PolymerLabs/cleankill/issues/4). After that I can delete this, and then eventually I can update to the version with promises.

File: src/browsers.ts (hunk 2 above)
Comment (rictic) [inline src/browsers.ts]: Done

File: src/browsers.ts (hunk 3 above)
Comment (rictic) [inline src/browsers.ts]: Done

File: src/browsers.ts (hunk 4 above)
Comment (rictic) [inline src/browsers.ts]: Nope, removed unnecessary jsdocs.

File: custom_typings/cleankill.d.ts (hunk 1 above)
Comment (justinfagnani) [inline custom_typings/cleankill.d.ts]: k

**Comment by justinfagnani** [comment]
LGTM

**Pull request merged**